README.md

```diff
@@ -1,2 +1,560 @@
+a homogeneous array of a basic type, or a CBOR "binary" type.
+Because `LLRP` Messages and Parameters are [highly structured][llrp_diagram],
+this device service maps Specs, Configuration, Capabilities, and Notifications 
+to and from JSON-encoded string using `Go`'s `json` package 
+and the structures defined in our [LLRP Library][llrp_library].
+
+When a Parameter or Message includes arbitrary data
+(e.g., the contents of a tag's `EPC` memory bank or a `CustomParameter` payload),
+they're represented in `Go` as a `[]byte`, which `Go` marshals and unmarshals 
+as strings representing the base64-encoded data.
+
+For requests to read a `deviceResource` (i.e., a `GET` request), 
+the service determines which `LLRP` message to send based upon the resource name.
+It marshals the result to JSON and returns it as a string EdgeX `Reading`.
+`LLRP` constants are encoded according to the `LLRP` spec 
+(e.g., the `StopTriggerType` of an `AISpec` is returned as 0, 1, or 2).
+The service uses only the resource name and ignores any attributes it may have;
+custom LLRP parameter extensions are not supported for resources read requests, 
+nor is the LLRP `CustomMessage` (Message Type 1023).
+
+The following list details the resource names the service recognizes 
+and how it satisfies the read request:
+
+- `ReaderCapabilities` sends `GET_READER_CAPABILITIES` (Message Type 1)
+    with `RequestedData: All`.
+    It returns the resulting `GET_READER_CAPABILITIES_RESPONSE` (Message Type 12).
+- `ReaderConfig` sends `GET_READER_CONFIG` (Message Type 2) 
+    with `RequestedData: All`, and `AntennaID`, `GPIPort`, and `GPOPort` set to 0.
+    It returns the resulting `GET_READER_CONFIG_RESPONSE` (Message Type 12).
+- `ROSpec` sends `GET_ROSPECS` (Message Type 26)
+    and returns `GET_ROSPECS_RESPONSE` (Message Type 36).
+- `AccessSpec` sends `GET_ACCESSSPECS` (Message Type 44)
+    and returns `GET_ACCESSSPECS_RESPONSE` (Message Type 44).
+    
+You can configure `deviceCommands` in your device profile
+to read more than one resource at a time,
+in which case the device service attempts each of the above requests in series
+and returns the results of all of them.
+
+For `LLRP` messages that require only an `ROSpecID` or `AccessSpecID`,
+we define them as operations upon `uint32` `deviceResource`s with those names. 
+(note that EdgeX requires passing these as strings when calling `deviceCommand`s)
+To disambiguate the desired write operation,
+`deviceCommands` that use them must specify them as the first resource to `set`,
+and must include a "pseudo-resource" called `Action`,
+a string which must be one of "Enable", "Disable", "Start", "Stop", or "Delete".
+Note that it is not possible in `LLRP` to start or stop an `AccessSpec`,
+so those only apply to `ROSpec`s.
+Because we must use this pseudo-resource to know what Action to take,
+It is not possible to write more than one `deviceResource` at a time.
```

Comment:
It isn't a danger of concurrent transactions -- the LLRP library is quite careful to about serializing concurrent requests, and the driver's wrapper even handles buffering messages during the initial connection and reconnects. 

I presume if you try to write one of the "action required" resources, it won't blow up, but you'll just get an error, since you'll be missing that parameter. At least, our code checks for it & reports an error, but I'm assuming that the SDK doesn't handle writes to a resource radically differently than a command, based on what you've written. I'm not sure exactly what the mapping is between the profile <-> SDK <-> Driver code. I've looked through it more than once, but there's a lot to trace through, so at the time, I experimented with different profiles to see what would or wouldn't eventually end up in our code. 

I seem to remember that I found you couldn't have a command that used the same resource more than once, but again, it's been quite a while, and I may be misremembering. If it is true, then if we replaced the `Action` resource with a resource-per-action and iterated through the `[]CommandRequest` parameter like the `HandleReadCommands` code does, it would allow some interaction patterns that aren't currently supported. For example, you'd be able to configure a `deviceCommand.set` with a `EnableROSpecID` resource and `StartROSpecID` resource each with the same `parameter` value. On the other hand, you still wouldn't be able to configure one with multiple `EnableROSpecID` resources that use different `parameter` values. When I first started writing device service code, I the SDK intended us to process requests by handling each `[]CommandRequest`, but that doesn't seem to fit with the way most other device services are implemented, so maybe it's better that we didn't.